database/sql/src/main/scala/cromwell/database/slick/tables/CustomLabelEntryComponent.scala

```diff
@@ -0,0 +1,52 @@
+package cromwell.database.slick.tables
+
+import cromwell.database.sql.tables.CustomLabelEntry
+import slick.model.ForeignKeyAction.Cascade
+
+trait CustomLabelEntryComponent {
+
+  this: DriverComponent with WorkflowMetadataSummaryEntryComponent =>
+
+  import driver.api._
+
+  class CustomLabelEntries(tag: Tag)
+    extends Table[CustomLabelEntry](tag, "CUSTOM_LABEL_ENTRY") {
+    def customLabelEntryId = column[Long]("CUSTOM_LABEL_ENTRY_ID", O.PrimaryKey, O.AutoInc)
+
+    def customLabelKey = column[String]("CUSTOM_LABEL_KEY", O.Length(255))
+
+    def customLabelValue = column[String]("CUSTOM_LABEL_VALUE", O.Length(255))
+
+    def workflowExecutionUuid = column[String]("WORKFLOW_EXECUTION_UUID", O.Length(100))
+
+    override def * = (customLabelKey, customLabelValue, workflowExecutionUuid,
+    customLabelEntryId.?) <> (CustomLabelEntry.tupled, CustomLabelEntry.unapply)
+
+    def fkCustomLabelEntryWorkflowExecutionUuid = foreignKey("FK_CUSTOM_LABEL_ENTRY_WORKFLOW_EXECUTION_UUID",
+      workflowExecutionUuid, workflowMetadataSummaryEntries)(_.workflowExecutionUuid, onDelete = Cascade)
+
+    def ucCustomLabelEntryClkClvWeu = index("UC_CUSTOM_LABEL_ENTRY_CLK_CLV_WEU",
+      (customLabelKey, customLabelValue, workflowExecutionUuid), unique = true)
+}
+
+  val customLabelEntries = TableQuery[CustomLabelEntries]
+
+  val customLabelEntryIdsAutoInc = customLabelEntries returning
+    customLabelEntries.map(_.customLabelEntryId)
+
+  val labelEntriesForWorkflowUuid = Compiled(
+    (workflowUuid: Rep[String]) => for {
+      customLabelEntry <- customLabelEntries
+      if customLabelEntry.workflowExecutionUuid === workflowUuid
+    } yield (customLabelEntry.customLabelKey, customLabelEntry.customLabelValue)
+  )
+
+  def existsWorkflowIdLabelKeyAndValue(workflowId: Rep[String])
+                                      (labelKeylabelValue: (String, String)): Rep[Boolean] = {
+    customLabelEntries.filter(customLabelEntry =>
+      customLabelEntry.workflowExecutionUuid === workflowId &&
+        customLabelEntry.customLabelKey === labelKeylabelValue._1 &&
+        customLabelEntry.customLabelValue === labelKeylabelValue._2
+    ).exists
```

Comment:
IOW balance how much of a PITA it is now vs someone trying to debug the issue completely blind a few months down the road